lib/publiccloud/images.pm

```diff
@@ -0,0 +1,115 @@
+use warnings;
+use testapi;
+use utils;
+use version_utils;
+use Mojo::Base -strict;
+use Mojo::UserAgent;
+
+our @EXPORT = qw(parse_image_regex get_latest_image_version);
+
+# Fetch links from a given url that match the given regexp
+sub fetch_matching_links {
+    my $url   = $_[0];
+    my $match = $_[1];
+    my $ua    = Mojo::UserAgent->new();
+    my $links = $ua->get($url)->res->dom->find('a')->map(attr => 'href');
+    return grep { $_ =~ $match } @{$links};
+}
+
+# Compare two version strings (e.g. '0.9.1' and '0.9.0' or '0.141' and '0.139')
+# returns -1 if the first is smaller, 1 if larger and 0 if they are the same
+sub version_cmp {
+    my @v1 = split('\.', $_[0]);
+    my @v2 = split('\.', $_[1]);
+
+    my $l = scalar @v1;
+    die "No versions to compare"  if ($l <= 0);
+    die "Version scheme mismatch" if ($l != scalar @v2);
+    for (my $i = 0; $i < $l; $i++) {
+        if ($v1[$i] < $v2[$i]) {
+            return -1;
+        } elsif ($v1[$i] > $v2[$i]) {
+            return 1;
+        }
+    }
+    return 0;
+}
+
+# Given an PC image, this routine returns the URL of the latest available version of this image
+sub get_latest_image_version {
+    my $image = $_[0];
+    my $regex = $_[1] // '^SLES(?<version>\d\d-SP\d)-(?<img_type>[^\.]+)?\.(?<arch>x86_64|aarch64)-(?<build_kiwi>[\d\.]+)-(?<url_flavor>.+-)?Build(?<build>[\d\.]+)\.(?<ext>tar.gz|raw.xz|vhdfixed.xz)$';
+    my $i        = rindex($image, '/');
+    my $filename = substr($image, $i + 1);
+    my $basepath = substr($image, 0, $i) . '/';
+
+
+    # Fetch links from the URL base path that match the generated regex
+    my @links = fetch_matching_links("$basepath", ".*$regex\$");
+    my $links = scalar @links;
+    die "No images found with given link and regex" if ($links <= 0);
+    print("Running PublicCloud image check on $links links ... \n");
+    print("    Image location       : $basepath\n");
+    print("    Generated regex      : $regex\n");
+    print("\n");
+
+
+    die "Error building regex for $filename" unless ($filename =~ $regex);
+    my $version    = $1;
+    my $type       = $2;
+    my $arch       = $3;
+    my $kiwi_build = $4;
+    my $flavor     = $5 // "";
+    my $build      = $6;
+    my $ext        = $7;
+
+    for my $link (@links) {
+        if ($link =~ $regex) {
+            my $c_version    = $1;
+            my $c_type       = $2;
+            my $c_arch       = $3;
+            my $c_kiwi_build = $4;
+            my $c_flavor     = $5 // "";
+            my $c_build      = $6;
+            my $c_ext        = $7;
+            # Skip non-matching stuff
+            next if $version ne $c_version || $type ne $c_type || $arch ne $c_arch || $flavor ne $c_flavor || $ext ne $c_ext;
+
+            # Compare versions
+            my $cmp = version_cmp($kiwi_build, $c_kiwi_build);
+            if ($cmp < 0) {
+                print("[ERR] Outdated kiwi-version detected: ($c_kiwi_build, $kiwi_build) < ($kiwi_build, $build) - $link\n");
```

Comment:
what about `record_info ` function instead ? it will give you red square in test which is more visible

Reply:
Unnecessary prints have been removed